+++ 
abstract = "An important criterion to consider in genetic evaluations is the extent of genetic connectedness across management units (MU), especially if they differ in their genetic mean. Reliable comparisons of genetic values across MU depend on the degree of connectedness; the higher the connectedness, the more reliable the comparison. Traditionally, genetic connectedness was calculated through pedigree-based methods; however, in the era of genomic selection, this can be better estimated utilizing new approaches based on genomics. Most procedures consider only additive genetic effects, which may not accurately reflect the underlying gene action of the evaluated trait, and little is known about the impact of non-additive gene action on connectedness measures. The objective of this study was to investigate the extent of genomic connectedness measures, for the first time, in Brazilian field data by applying additive and non-additive relationship matrices using a fatty acid profile dataset from seven farms located in the three regions of Brazil, which are part of the three breeding programs. Myristic acid (C14:0) was used due to its importance for human health, and reported presence of non-additive gene action. The pedigree included 427,740 animals and 925 of them were genotyped using the Bovine high-density genotyping chip. Six relationship matrices were constructed, parametrically and non-parametrically capturing additive and non-additive genetic effects from both pedigree and genomic data. We assessed genome-based connectedness across MU using the prediction error variance of difference (PEVD) and the coefficient of determination (CD). PEVD values ranged from 0.540 - 1.707, and CD from 0.146 - 0.456. Genomic information consistently enhanced the measures of connectedness compared to the numerator relationship matrix by at least 63%. Combining additive and non-additive genomic kernel relationship matrices or a non-parametric relationship matrix increased the capture of connectedness. Overall, the Gaussian kernel yielded the largest measure of connectedness. Our findings showed that connectedness metrics can be extended to incorporate genomic information and non-additive genetic variation using field data. We propose that different genomic relationship matrices can be designed to capture additive and non-additive genetic effects, increase the measures of connectedness, and to more accurately estimate the true state of connectedness in herds."
abstract_short = ""
authors = ["Sabrina T. Amorim", "__Haipeng Yu__", "Mehdi Momen", "Lúcia Galvão de Albuquerque", "Angélica S. Cravo Pereira", "Fernando Baldi", "and Gota Morota"]
date = "2020-09-02"
# image = ""
# image_preview = ""
math = false
publication = "Journal of Animal Science."
publication_short = "Journal of Animal Science"
publication_types = ["2"]
selected = false
title = "An assessment of genomic connectedness measures in Nellore cattle"
url_code = ""
url_dataset = ""
# url_pdf = "papers/GC2.pdf"
url_project = ""
url_slides = ""
url_video = ""

[[url_custom]]
name = "DOI"
url = "https://academic.oup.com/jas/advance-article/doi/10.1093/jas/skaa289/5900679"

[[url_custom]]
name = "PubMed"
url = "https://pubmed.ncbi.nlm.nih.gov/32877515/"

[[url_custom]]
name = "EuropePMC"
url = "http://europepmc.org/article/MED/32877515"

+++
