<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
  <head>
    <title>Bayesian confirmatory factor analysis and Bayesian network for high-dimensional phenotypic data</title>
    <meta charset="utf-8" />
    <meta name="author" content="Haipeng Yu  https://haipengu.github.io/   Gota Morota   http://morotalab.org/" />
    <link href="libs/remark-css/default.css" rel="stylesheet" />
    <link href="libs/remark-css/default-fonts.css" rel="stylesheet" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

# Bayesian confirmatory factor analysis and Bayesian network for high-dimensional phenotypic data
## GWAS Workshop <span class="citation">@VT</span>
### Haipeng Yu <br /><a href="https://haipengu.github.io/" class="uri">https://haipengu.github.io/</a> <br /> Gota Morota <br /> <a href="http://morotalab.org/" class="uri">http://morotalab.org/</a>
### 2019/6/26

---





# Motivation: High dimensional and diverse phenotypes
&lt;img src="Pixelomics.png" height="400px" width="750px"/&gt;

---
# How to handle a large number of phenotypes?
More and more phenotypes are being generated across time and space
&lt;img src="bigPhe.png" height="130px" width="600px"/&gt;

Challenges:
- high dimensional phenotypes
- diverse phenotypes
- how to make sense of these data and interpret
- multi-trait linear mixed model is computationally challenging

Objective:
- leverage Bayesian confirmatory factor analysis and Bayesian network to characterize a wide spectrum of rice phenotypes

---
# Bayesian confirmatory factor analysis
Assume observed phenotypes are derived from underlying latent variables

`\begin{align*}
    \mathbf{T} =  \mathbf{\Lambda} \mathbf{F} + \mathbf{s}
\end{align*}`
  
- `\(\mathbf{T}\)` is the `\(t \times n\)` matrix of observed phenotypes (413 accessions)

- `\(\mathbf{\Lambda}\)` is the `\(t \times q\)` factor loading matrix

- `\(\mathbf{F}\)` is the `\(q \times n\)` latent variables matrix

- `\(\mathbf{s}\)` is the `\(t \times n\)` matrix of specific effects.

Variance-covariance model: 
`\begin{align*}
    var\mathbf{(T)} &amp;= \mathbf{\Lambda}\mathbf{\Phi}\mathbf{\Lambda}' + \mathbf{\Psi},
\end{align*}`


- `\(\mathbf{\Phi}\)` is the variance of latent variables

- `\(\mathbf{\Psi}\)` is the variance of specific effects

---
# Prior distributions 

`\begin{align*}
    var\mathbf{(T)} &amp;= \mathbf{\Lambda}\mathbf{\Phi}\mathbf{\Lambda}' + \mathbf{\Psi},
\end{align*}`

- Factor loading matrix: 
`\begin{align*}
&amp; \mathbf{\Lambda} \sim \mathcal{N}(0, 0.01)
\end{align*}`

- Variance of latent variables:
`\begin{align*}
\mathbf{\Phi} \sim \mathcal{W}^{-1}(\mathbf{I}_{66}, 7) 
\end{align*}`

- Variance of specific effects:
`\begin{align*}
\mathbf{\Psi} \sim \Gamma^{-1} (1, 0.5)
\end{align*}`
---
# Define 6 latent variables from 48 phenotypes


1. Grain Morphology (Grm, 11)
    - Seed length (Sl), Seed width (Sw), Seed volume (Sv), etc
2. Morphology (Mrp, 14)
    - Flag leaf length (Fll), Flag leaf width (Flw), etc
3. Flowering Time (Flt, 7)
    - Flowering time in Arkansas (Fla), Flowering time in Aberdeen (Flb), etc
4. Ionic components of salt stress (Iss, 6) 
    - Na shoot (Nas), K shoot salt (Kss), etc
5. Yield (Yid, 5)
    - Panicle number per plant (Pnu),   Panicle length (Pal), etc
6. Morphological salt response (Msr, 5)  
    - Shoot BM ratio (Sbr), Root BM ratio (Rbr), etc

---
# Study the genetics of each latent variable
&lt;img src="BCFA.jpg" height="530px" width="700px"/&gt;

---
# Multivariate analysis
Bayesian genomic best linear unbiased prediction
    - separate genetic effects from noise (44K SNPs)
    
`\begin{align*}
    \mathbf{F} = \boldsymbol{\mu} + \mathbf{Xb} + \mathbf{Zu} + \boldsymbol{\epsilon}
\end{align*}`

&lt;div align="center"&gt;    
&lt;img src="MTM.png" height="120px" width="500px"/&gt;
&lt;/div&gt;

- `\(\mathbf{F}\)` : Vector of factor scores
- `\(\mathbf{X}\)` : Incidence matrix for fixed effects 
- `\(\mathbf{Z}\)` : Incidence matrix for additive genetic effects
- `\(\mathbf{b}\)`: Vector of fixed effects
- `\(\mathbf{u}\)`: Vector of additive genetic effects
- `\(\mathbf{e}\)`: Vector of residuals

---
# Piror distributions
`\begin{align*}
    \mathbf{F} = \boldsymbol{\mu} + \mathbf{Xb} + \mathbf{Zu} + \boldsymbol{\epsilon}
\end{align*}`

&lt;div align="center"&gt;    
&lt;img src="MTMvar.png" height="100px" width="500px"/&gt;
&lt;/div&gt;
- `\(\mathbf{\mu}\)`, `\(\mathbf{b}\)` were assigned a flat prior

- `\(\boldsymbol{\Sigma_{u}}\)`, `\(\boldsymbol{\Sigma_{e}}\)` are variance-covariance matrix between latent variables


`\begin{align*}
\boldsymbol{\Sigma_{u}}, \boldsymbol{\Sigma_{e}} \sim \mathcal{W}^{-1}(\mathbf{I}_{66}, 6) 
\end{align*}`


---
# Bayesian Network
Probabilistic directed acyclic graphical model (DAG)
    - interrelationships (Edges) among latent variables (Nodes)
    - genetic selection for breeding requires causal assumptions

&lt;div align="center"&gt; 
&lt;img src="BCFA2.png" height="256px" width="400px"/&gt;
&lt;/div&gt;
---
# Constraint-based learning
&lt;img src="Fig3.jpg" height="500px" width="700px"/&gt;

---
# Score-based learning
&lt;img src="Score_based_algorithm.png" height="500px" width="700px"/&gt;

---
# Examples of algorithms 
- Score-based algorithm
  - Hill climbing 
  - Tabu search

- Hybrid algorithm
  - Max-Min Hill Climbing algorithm
  - General 2-Phase Restricted Maximization algorithm
---
# Standardized factor loadings
&lt;div align="center"&gt; 
&lt;img src="load2.png" height="506px" width="500px"/&gt;
&lt;/div&gt;
---
# Standardized factor loadings
&lt;div align="center"&gt; 
&lt;img src="load3.png" height="466px" width="500px"/&gt;
&lt;/div&gt;

---
# Genetic correlations among latent variable
&lt;div align="center"&gt; 
&lt;img src="Cor_plot_for_presentation.png" height="530px" width="700px"/&gt;
&lt;/div&gt;
---
# Hill Climbing algorithm
&lt;img src="BN_HC.png" height="530px" width="700px"/&gt;




---
# Tabu algorithm
&lt;img src="BN_TABU.png" height="530px" width="700px"/&gt;



---
# Max-Min Hill Climbing algorithm
&lt;img src="BN_MMHC.png" height="530px" width="700px"/&gt;



---
# General 2-Phase Restricted Maximization algorithm
&lt;img src="BN_RSMAX2.png" height="460px" width="650px"/&gt;


---
# Consensus Bayesian network
&lt;img src="BN_consensus.png" height="460px" width="650px"/&gt;

---
# FA vs. PCA

- What is the main difference between principal component analysis (PCA) and factor analysis (FA)? 

- Confirmatory factor analysis (CFA) vs. Explanatory factor analysis (EFA)

---
# CFA vs. EFA
&lt;div align="center"&gt;
&lt;img src="EFA_CFA.png" height="470px" width="750px"/&gt;
&lt;/div&gt;

&lt;!-- # CFA vs. EFA --&gt;
&lt;!-- Wheat Data structure (CFA) : --&gt;
&lt;!-- &lt;div align="center"&gt;  --&gt;
&lt;!-- &lt;img src="Data_Structure.png" height="500px" width="650px"/&gt; --&gt;
&lt;!-- &lt;/div&gt; --&gt;

&lt;!-- --- --&gt;
&lt;!-- # CFA vs. EFA --&gt;
&lt;!-- Bayesian Network (CFA) --&gt;

&lt;!-- --- --&gt;
&lt;!-- # CFA vs. EFA --&gt;
&lt;!-- Number of factors in EFA:  --&gt;
&lt;!-- &lt;div align="center"&gt;  --&gt;
&lt;!-- &lt;img src="Scree_Plot.png" height="500px" width="650px"/&gt; --&gt;
&lt;!-- &lt;/div&gt; --&gt;

&lt;!-- --- --&gt;
&lt;!-- # CFA vs. EFA --&gt;
&lt;!-- Bayesian Network of EFA:  --&gt;

&lt;!-- --- --&gt;
&lt;!-- # CFA vs. EFA --&gt;
&lt;!-- Factor loading of EFA: --&gt;

---
# Summary

- Bayesian cofirmatory factor analysis allows to work at the
level of latent variables

- Bayesian network can be applied to predict the potential influence of external interventions or selection associated with target traits

- Provide greater insights than pairwise-association measures of multiple phenotypes

- It is possible to dissect genetic signals from high-dimensional phenotypes if we focus on underlying patterns in big data

---
# Useful links
&lt;div align="center"&gt;
&lt;img src="G3paper.png" height="300px" width="600px"/&gt;
&lt;/div&gt;

- Genomic Bayesian confirmatory factor analysis and Bayesian network to characterize a wide spectrum of rice phenotypes (https://www.g3journal.org/content/9/6/1975)

- An illustrative example of Bayesian Network (https://haipengu.github.io/Rmd/GBN.html)
    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();</script>

<script>
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
